libs/model/src/aggregates/token-allocation/magnaSync.ts

```diff
@@ -0,0 +1,95 @@
+import { logger } from '@hicommonwealth/core';
+import { models } from '@hicommonwealth/model/db';
+import { delay } from '@hicommonwealth/shared';
+import { QueryTypes } from 'sequelize';
+
+const log = logger(import.meta);
+
+export type TokenAllocationSyncArgs = {
+  user_id: number;
+  user_name: string;
+  user_email: string;
+  wallet_address: string;
+  token_allocation: number;
+};
+
+/**
+ * Callback Response.
+ *
+ * Should return id if allocation was created or already exists.
+ *
+ * - id: Allocation ID
+ * - error: Error message if failed to create allocation
+ */
+export type TokenAllocationSyncResponse = { id?: string; error?: string };
+
+export async function magnaSync(
+  apiCallback: (
+    args: TokenAllocationSyncArgs,
+  ) => Promise<TokenAllocationSyncResponse>,
+  batchSize = 10,
+  breatherMs = 1000,
+) {
+  try {
+    let found = true;
+    while (found) {
+      await delay(breatherMs); // take a breath to keep rate limit at BATCH_SIZE/sec
+
+      // Load next batch of allocations to sync with Magna
+      const batch = await models.sequelize.query<TokenAllocationSyncArgs>(
+        `
+          SELECT
+            A.user_id,
+            U.profile->>'name' as user_name,
+            U.profile->>'email' as user_email,
+            C.address as wallet_address,
+            A.token_allocation
+          FROM
+            "HistoricalAllocations" A
```

Comment:
There is also the "AuraAllocations" table.

Reply:
Is this another sync category? Let's discuss tomorrow.